src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -317,9 +317,21 @@ def _dispatchWflow(self, wflow):
             #       {REPLICATING, STUCK, SUSPENDED, WAITING_APPROVAL}.)
             #       We still need some proper logging for them.
             msg = "Skipping workflow: %s - tape transfers are not yet completed." % wflow['RequestName']
-            msg += " Will retry again in the next cycle."
+            msg += " Will retry again in the next cycle. deleting the MS-Transferor rules"
             self.logger.info(msg)
             self._checkStatusAdvanceExpired(wflow, additionalInfo=msg)
+            for pline in self.mstrlines:
+                try: 
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline while cleaning Transfer rukes. Workflow: %s. Error:  \n%s. " % (pline.name, wflow['RequestName'], str(ex))
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg)
+                    self._checkStatusAdvanceExpired(wflow, additionalInfo=msg)
+                    continue
+                if wflow['CleanupStatus'][pline.name]:
```

Comment:
If you update the cleanup counters, while only calling a subset of the cleanup lines, on the next stage, while traversing the archival pipeline, the workflow will be considered cleaned and will trigger a premature archival, before having it passed through all the cleanup lines. This is a consequence from the fact that  the `archival` step and the `cleanup` steps live in the same micro service and are called sequentially, but that's another story and we can live with what we have already. 

Anyway, since you are calling a partial and in advance cleanup of only input rules, the way to avoid this is by NOT updating the cleanup counters. So please just remove these two lines here.